src/v0/analytics.ts

```diff
@@ -202,11 +217,20 @@ export class Analytics {
     })
   }
 
-  getVouchersReports(): Promise<AnalyticsResponse> {
+  getVouchersReports(query?: VoucherOptions): Promise<AnalyticsResponse> {
     return new Promise(async (resolve, reject) => {
       try {
-        const uri = `${this.options.base}${this.endpoint}/${this.options.user}/reports/vouchers`
+        let uri = `${this.options.base}${this.endpoint}/${this.options.user}/reports/vouchers`
+        const q = query || {}
+        const queryKeys = Object.keys(q)
+          .map((key: string) => {
+            return `${key}=${q[key].toString()}`
+          })
+          .join('&')
```

Comment:
There is a library for making querystrings that we use called 'qs'. It makes the code simpler, it would look like this:
const queryString = qs.stringify(query)

Reply:
I like this